### PR TITLE
Issue 783 - 'hzn register' handle node pattern defined in the exchange

### PR DESCRIPTION
**Pull request by linggao**
Did the pattern checking for both hzn and anax api because we would like to give proper messages to the user who are using `hzn register` in some special cases.

### CODE REVIEW

File: cli/cliutils/cliutils.go [hunk 1]
```diff
@@ -327,6 +327,19 @@ func TrimOrg(org, id string) (string, string) {
...
 
+// Add the given org to the id if the id does not already contain an org
+func AddOrg(org, id string) string {
+	substrings := strings.Split(id, "/")
+	if len(substrings) <= 1 { // this means id was empty, or did not contain '/'
+		return fmt.Sprintf("%v/%v", org, id)
+	} else if len(substrings) == 2 {
+		return id
+	} else {
+		Fatal(CLI_INPUT_ERROR, i18n.GetMessagePrinter().Sprintf("the resource id can not contain more than 1 '/'"))
```
Comment (dabooz) [inline cli/cliutils/cliutils.go]: Why does the message say "the resource id can not" instead of "the id can not". What is the word resource referring to?

Comment (linggao) [reply cli/cliutils/cliutils.go]: It is copied from the function TrimOrg() above the AddOrg().  TrimOrg is called in many places. The id passed in could be a pattern id, node id, service id. I guess the old code uses `resource id` as a general term. Shall I change "resource id" in both places to just "id"?

Comment (dabooz) [reply cli/cliutils/cliutils.go]: So yes, I think we should remove the word resource from both places

Comment (linggao) [reply cli/cliutils/cliutils.go]: Done.

**Pull request merged**